bool initialized_
time data